pkg/ovs/ovsctl/appctl.go

```diff
@@ -75,18 +76,30 @@ func (c *ovsCtlClient) Trace(req *TracingRequest) (string, error) {
 	}
 
 	if req.SrcIP != nil {
-		if strings.Contains(req.Flow, "nw_src=") {
-			return "", newBadRequestError("duplicated 'nw_src' in flow")
+		nwSrcKey := getNwSrcKey(req.SrcIP)
+		if strings.Contains(req.Flow, fmt.Sprintf("%s=", nwSrcKey)) {
+			return "", newBadRequestError(fmt.Sprintf("duplicated '%s' in flow", nwSrcKey))
 		} else {
-			nwSrc = fmt.Sprintf("nw_src=%s,", req.SrcIP.String())
+			nwSrc = fmt.Sprintf("%s=%s,", nwSrcKey, req.SrcIP.String())
+		}
+		if strings.Contains(nwSrcKey, "ipv6") {
```

Comment:
This looks indirect. Why not just check req.SrcIP is v4 or v6??

Reply:
req.SrcIP is used to decide `nwSrcKey`, it might be duplicated to use it to decide ipKey. I am thinking to change the return value of function `getNwSrcKey`, and include both ipKey and nwSrcKey, what do you think?